synapse/storage/databases/main/search.py

```diff
@@ -644,7 +650,8 @@ async def search_rooms(
         else:
             raise Exception("Unrecognized database engine")
 
-        args.append(limit)
+        # mypy expects to append only a `str`, not an `int`
+        args.append(limit)  # type: ignore[arg-type]
```

Comment:
It would be better to define the type of args at the first site it defined (line 558?) as `List[Union[str, int]]`.

Reply:
If I am change this:
```
synapse/storage/databases/main/search.py:611: error: Incompatible types in assignment (expression has type "List[str]", variable has type "List[Union[str, int]]")  [assignment]
synapse/storage/databases/main/search.py:611: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
synapse/storage/databases/main/search.py:611: note: Consider using "Sequence" instead, which is covariant
```

And if I am use `Sequence`:
```
synapse/storage/databases/main/search.py:582: error: "Sequence[Any]" has no attribute "append"  [attr-defined]
synapse/storage/databases/main/search.py:603: error: "Sequence[Any]" has no attribute "extend"  [attr-defined]
synapse/storage/databases/main/search.py:612: error: Unsupported operand types for + ("List[str]" and "Sequence[Any]")  [operator]
synapse/storage/databases/main/search.py:638: error: Unsupported operand types for + ("List[str]" and "Sequence[Any]")  [operator]
synapse/storage/databases/main/search.py:664: error: Argument 1 to "append" of "list" has incompatible type "int"; expected "str"  [arg-type]
```

Reply:
I'll try to take a look at this tomorrow, if I get time